Rover/GCS_Mavlink.cpp

```diff
@@ -185,6 +185,65 @@ void GCS_MAVLINK_Rover::send_rangefinder() const
         distance,
         voltage);
 }
+
+void GCS_MAVLINK_Rover::send_water_depth() const
+{
+    if (!HAVE_PAYLOAD_SPACE(chan, WATER_DEPTH)) {
+        return;
+    }
+
+    // only send for boats:
+    if (!rover.is_boat()) {
+        return;
+    }
+
+    RangeFinder *rangefinder = RangeFinder::get_singleton();
+
+    if (rangefinder == nullptr) {
+        return;
+    }
+
+    // depth can only be measured by a downward-facing rangefinder:
+    if (!rangefinder->has_orientation(ROTATION_PITCH_270)) {
+        return;
+    }
+
+    // get position
+    const AP_AHRS &ahrs = AP::ahrs();
+    Location loc;
+    IGNORE_RETURN(ahrs.get_location(loc));
+
+    for (uint8_t i=0; i<rangefinder->num_sensors(); i++) {
```

Comment:
Should we cycle backends? Before this fix I have have the link be completely flooded on a vehicle with several rangefinders.

Reply:
Yes, we should.

See last commit on here: https://github.com/ArduPilot/ardupilot/pull/28877